app/controllers/comments_controller.rb

```diff
@@ -1,10 +1,12 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
-  before_action :is_an_authorized_user, only: [:destroy, :create]
+  before_action :authorize_comment, except: [:index, :new, :create]
+  after_action :authorize_comment, only: [:new, :create]
 
   # GET /comments or /comments.json
   def index
     @comments = Comment.all
+    authorize @comments
```

Comment:
Nitpik: could remove this line if `authorize_comment` also handled @comments like so:

```ruby
    def authorize_comment
      authorize (@comment || Comment)
    end
```

Then you'd include `:index` in the `before_action` as well. But not a requirement for the assignment.